src/Extension/FluentSearchFormExtension.php

```diff
@@ -0,0 +1,94 @@
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\SS_List;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * Simple extension to enable SS Core search to work with SS Fluent
+ */
+class FluentSearchFormExtension extends SearchForm
+{
+    /**
+     * Return dataObjectSet of the results using current request to get info from form.
+     * Wraps around {@link searchEngine()}.
+     *
+     * @return SS_List
+     */
+    public function getResults()
+    {
+        // Get request data from request handler
+        $request = $this->getRequestHandler()->getRequest();
+
+        $keywords = $request->requestVar('Search');
+
+        $andProcessor = function ($matches) {
+            return ' +' . $matches[2] . ' +' . $matches[4] . ' ';
+        };
+
+        $notProcessor = function ($matches) {
+            return ' -' . $matches[3];
+        };
+
+        $keyword_patterns = [
+            'andProcessor' => '/()("[^()"]+")( and )("[^"()]+")()/i',
+            'andProcessor' => '/(^| )([^() ]+)( and )([^ ()]+)( |$)/i',
+            'notProcessor' => '/(^| )(not )("[^"()]+")/i',
+            'notProcessor' => '/(^| )(not )([^() ]+)( |$)/i',
+        ];
+
+        foreach ($keyword_patterns as $callback => $pattern) {
+            $keywords = preg_replace_callback($pattern, $$callback, $keywords);
+        }
+
+        $keywords = DB::get_conn()
+            ->escapeString(
+                $this->addStarsToKeywords($keywords)
+            );
+
+        $keywords_strrpl = str_replace(['*', '+', '-'], '', $keywords);
+        $htmlEntityKeywords = htmlentities($keywords, ENT_NOQUOTES, 'UTF-8');
+
+        $sitetreetables = '"SiteTree_Live"."Title", "SiteTree_Live"."MenuTitle", "SiteTree_Live"."Content", "SiteTree_Live"."MetaDescription"';
+        $current_locale = FluentState::singleton()->getLocale();
+        $sitetree_table_localised = "SiteTree_Localised_{$current_locale}";
+
+        $sql = <<<SQL
+            SELECT DISTINCT "SiteTree_Live"."ID", "SiteTree_Live"."Title", MATCH ({$sitetreetables}) AGAINST (?) AS Relevance
+            FROM "SiteTree_Live"
+            LEFT JOIN "SiteTree_Localised" AS "{$sitetree_table_localised}"
+            ON "SiteTree_Live"."ID" = "{$sitetree_table_localised}"."RecordID" AND "{$sitetree_table_localised}"."Locale" = ? AND (("{$sitetree_table_localised}"."ID" IS NOT NULL))
+            WHERE (MATCH ({$sitetreetables}) AGAINST (? IN BOOLEAN MODE) + MATCH ({$sitetreetables}) AGAINST (? IN BOOLEAN MODE) AND "SiteTree_Live"."ShowInSearch" <> 0)
+            ORDER BY Relevance DESC
+            LIMIT 20
+SQL;
+        $params = [
+            $keywords_strrpl,
+            $current_locale,
+            $keywords,
+            $htmlEntityKeywords
+        ];
+        $custom_results = DB::prepared_query($sql, $params);
+
+        $results = new ArrayList();
+
+        foreach ($custom_results as $record) {
+            $st = SiteTree::get_by_id($record['ID']);
```

Comment:
You could have a quicker result by putting it into one query.

```php
$ids = $custom_results->column('ID');
if ($ids) {
    $pages = SiteTree::get()->filter('ID', $ids);
} else {
    return null;
}
```

Reply:
Just saw this comment. Thanks for the suggestion, I'll make the update.

Reply:
Change made and pushed.